src/Components/Facility/FacilityCreate.tsx

```diff
@@ -243,6 +244,34 @@ export const FacilityCreate = (props: FacilityProps) => {
     },
   );
 
+  const facilitiesQuery = useQuery(routes.getAllFacilities);
+  const hubsQuery = useQuery(routes.getFacilityHubs, {
+    pathParams: {
+      id: facilityId!,
+    },
+    prefetch: !!facilityId,
+    onResponse: (res) => {
+      setHubFacilities(res.data?.results.map((d) => d.hub.id) as string[]);
+    },
+  });
+  const createHub = async (hubFacilityId: string) =>
+    await request(routes.createFacilityHub, {
+      body: {
+        hub_id: hubFacilityId,
+      },
+      pathParams: {
+        id: facilityId || "",
+      },
+    });
+  const deleteHub = async (hubFacilityId: string) =>
+    await request(routes.deleteFacilityHub, {
+      pathParams: {
+        id: facilityId || "",
+        hub_id: hubFacilityId,
+      },
+    });
+  const { data: facilities } = facilitiesQuery;
```

Comment:
Would this return "ALL" facilities? It'd have a per-page limit right?

Reply:
Also, shouldn't it be data.results instead? since data would be a paginated result